### PR TITLE
feat: `Must` assertions support

### LINKED ISSUE #459: feature request: add `stop-on-failure` support at steps level
Currently the `stop-on-failure` seems to be only available at test-case level, but it would be nice to have them also at steps level.

Use case would be for multi-steps (both test and custom executor level) where one step depends on another one:
```yml
- steps:
  - type: exec 
    script: exit 1
    assertions:
      - result.code ShouldEqual 0
    vars:
      stdout:
        from: result.systemout
  - type: exec 
    script: echo '{{.stdout}}'
    assertions:
      - result.code ShouldEqual 0
```

### PR DESCRIPTION
**Pull request by lowlighter**
Closes #459

Add support to `Must` assertions

Example:
```yml
- steps:
  - type: exec
    script: exit 1
    assertions:
      - result.code MustEqual 0
  # Remaining steps in this context will not be executed
```

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#698.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#698.1 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#705.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

**Comment by fsamin** [comment]
Hi,
Many thanks for your PR.
I don't like very much this new property `stop-on-failure: true`, My suggestion for this feature would be to handle it through the assertions.

I think we could handle both `result.code ShouldEqual 0` and `result.code MustEqual 0`.
**Should** and **Must** would have a different behavior like we have in `testify` lib with `assert.Equals` and `require.Equals`.

If a **Must** assertions fails : the whole testcase and testsuite would fail immediately.

What do you think ?

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#707.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

**Comment by lowlighter** [comment]
Yeah it's a good idea, it would actually solve #458 at the same time
I'll look into it :+1:

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#710.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

**Review by fsamin — APPROVED** [review]

File: process_testsuite.go [hunk 1]
```diff
@@ -44,6 +44,14 @@ func (v *Venom) runTestSuite(ctx context.Context, ts *TestSuite) {
...
 	}
 
+	exePath, err := os.Executable()
+	if err != nil {
+		log.Errorf("failed to get executable path: %v", err)
+	} else {
+		ts.Vars.Add("venom.executable", exePath)
+	}
+
+	ts.Vars.Add("venom.libdir", v.LibDir)
```
Comment (yesnault) [inline process_testsuite.go:54]: Can you add venom.libdir and venom.executable to "Builtin variables" on README please?

**Review by yesnault — CHANGES_REQUESTED** [review]

File: process_testsuite.go (hunk 1 above)
Comment (lowlighter) [reply process_testsuite.go:54]: done :+1:

**Review by yesnault — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#712.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

**Pull request merged**